File: antigravity/a-vanilla/run-3/src/orm/Book.php
<?php

use Tina4\ORM;

class Book extends ORM
{
    public string $tableName = "books";
    public string $primaryKey = "id";
    public bool $autoMap = true;

    public int $id;
    public string $title;
    public string $author;
    public int $publishedYear;
    public string $isbn;
    public ?string $coverImage = null;
    public string $createdAt;
    public string $updatedAt;

    /**
     * Get borrowing history for this book
     * @return Loan[]
     */
    public function loans(): array
    {
        return $this->hasMany(Loan::class, "book_id");
    }

    /**
     * Check if the book is currently available for borrowing
     * @return bool
     */
    public function isAvailable(): bool
    {
        // Search for active loans where returned_date is null
        $activeLoans = (new Loan())->where("book_id = ? AND (returned_date IS NULL OR returned_date = '')", [$this->id]);
        return count($activeLoans) === 0;
    }
}

File: antigravity/a-vanilla/run-3/src/orm/book.py
from tina4_python.orm import ORM, IntegerField, StringField, has_many

class Book(ORM):
    table_name = "books"

    id = IntegerField(primary_key=True, auto_increment=True)
    title = StringField(required=True)
    author = StringField(required=True)
    published_year = IntegerField(required=True)
    isbn = StringField(required=True)
    cover_image = StringField(default="")

    # Relationship
    loans = has_many("Loan", foreign_key="book_id")

    @property
    def is_available(self) -> bool:
        from src.orm.loan import Loan
        active_loans = Loan.where("book_id = ? AND returned = 0", [self.id])
        return len(active_loans) == 0

    @property
    def active_loan(self):
        from src.orm.loan import Loan
        active_loans = Loan.where("book_id = ? AND returned = 0", [self.id])
        return active_loans[0] if active_loans else None
